zetasql/fuzzing/zetasql_expression_proto_to_string.cc

```diff
@@ -0,0 +1,146 @@
+//
+// Copyright 2020 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include <string>
+
+#include "zetasql/fuzzing/zetasql_expression_grammar.pb.h"
+
+using zetasql_expression_grammar::Expression;
+using zetasql_expression_grammar::LiteralExpr;
+using zetasql_expression_grammar::IntegerLiteral;
+using zetasql_expression_grammar::NumericLiteral;
+using zetasql_expression_grammar::CompoundExpr;
+using zetasql_expression_grammar::BinaryOperation;
+
+#define TO_STRING(TYPE, VAR_NAME) std::string TYPE##ToString(const TYPE& VAR_NAME)
+
+namespace zetasql_fuzzer {
+
+// Forward declaration
+TO_STRING(Expression, expr);
+
+TO_STRING(LiteralExpr, lit_expr);
+TO_STRING(IntegerLiteral, integer);
+TO_STRING(NumericLiteral, numeric);
+
+TO_STRING(CompoundExpr, comp_expr);
+TO_STRING(BinaryOperation, operation);
+
+inline void HandleUndefined(const std::string& error) {
+  std::cerr << error << std::endl;
+  std::abort();
+}
+
+inline std::string HandleDefault() {
+  return "0";
+}
+
+inline std::string Padded(const std::string& s) {
+  return " " + s + " ";
```

Comment:
Nit:  Might be very slightly faster to use absl::StrCat here.

Reply:
This is a primitive interface to be improved upon later. We'll likely include variable-length white-space characters in between.